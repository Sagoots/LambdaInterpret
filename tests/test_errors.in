( (\x.x)
\x.
1 2
#define BAD_MACRO_1
#define BAD_MACRO_2 =
#define 123 = (\x.x)
UNDEFINED_MACRO A
\x. (x x