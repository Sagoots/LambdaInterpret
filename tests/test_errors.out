--- λ-Calculus REPL ---
Input from: C:\Users\solom\Documents\Wolfram\lambdaREPL\tests\test_errors.in
Output to: C:\Users\solom\Documents\Wolfram\lambdaREPL\tests\test_errors.out
Enter expressions or macro definitions (e.g., #define ID <expression>).
 Type 'exit' or 'quit' to end.
λ> Parsing error: Expected MyLambdaInterpreter`Private`closeBrac, found MyLambdaInterpreter`Private`EOF

λ> Syntax error: Unexpected token MyLambdaInterpreter`Private`EOF at atom parsing

λ> 1 2 →
(λf x.f x) 2 →
λx.2 x →
2 →
λf x.f (f x)

λ> Invalid macro definition format. Expected: #define <ID> <expression>


λ> Syntax error: Unexpected token MyLambdaInterpreter`Private`equal at atom parsing

λ> Invalid macro definition format. Expected: #define <ID> <expression>


λ> UNDEFINED_MACRO A

λ> Parsing error: Expected MyLambdaInterpreter`Private`closeBrac, found MyLambdaInterpreter`Private`EOF

λ> End of input or input canceled. Exiting REPL.
